tools/rtpsrelay/GuidAddrSet.h

```diff
@@ -10,15 +10,27 @@
 #include <dds/DCPS/TimeTypes.h>
 #include <dds/DCPS/RTPS/RtpsDiscovery.h>
 
+#include <ace/INET_Addr.h>
+
 namespace RtpsRelay {
 
-typedef std::map<AddrPort, OpenDDS::DCPS::MonotonicTimePoint> AddrSet;
+struct PortSet {
+  std::map<u_short, OpenDDS::DCPS::MonotonicTimePoint> spdp_ports, sedp_ports, data_ports;
+  bool empty() const { return spdp_ports.empty() && sedp_ports.empty() && data_ports.empty(); }
+};
+
+struct InetAddrHash {
+  std::size_t operator()(const ACE_INET_Addr& addr) const noexcept
+  {
+    return addr.hash();
```

Comment:
Just a heads up... this hashing is not great, especially for ipv6.

Reply:
I'm basing it on `typedef std::unordered_map<Remote, OpenDDS::DCPS::GUID_t, RemoteHash> RemoteMap` in the existing code.  Maybe they could both be improved in a future PR.